### PR TITLE
Update changelog to note changes to http request/response size metric names

### PR DESCRIPTION
**Pull request by dashpole**
Address https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4707#pullrequestreview-1872246084

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The fallback options in  `go.opentelemetry.io/contrib/exporters/autoexport` now accept factory functions. (#4891)
   - `WithFallbackMetricReader(metric.Reader) MetricOption` is replaced with `func WithFallbackMetricReader(func(context.Context) (metric.Reader, error)) MetricOption`.
   - `WithFallbackSpanExporter(trace.SpanExporter) SpanOption` is replaced with `WithFallbackSpanExporter(func(context.Context) (trace.SpanExporter, error)) SpanOption`.
+- The `http.server.request_content_length` metric in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` is changed to `http.server.request.size`.(#4707)
+- The `http.server.response_content_length` metric in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` is changed to `http.server.response.size`.(#4707)
```
Comment (StarpTech) [inline CHANGELOG.md:36]: Can we also update the github release notes? That's my source all the time :smile:

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The fallback options in  `go.opentelemetry.io/contrib/exporters/autoexport` now accept factory functions. (#4891)
   - `WithFallbackMetricReader(metric.Reader) MetricOption` is replaced with `func WithFallbackMetricReader(func(context.Context) (metric.Reader, error)) MetricOption`.
   - `WithFallbackSpanExporter(trace.SpanExporter) SpanOption` is replaced with `WithFallbackSpanExporter(func(context.Context) (trace.SpanExporter, error)) SpanOption`.
+- The `http.server.request_content_length` metric in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` is changed to `http.server.request.size`.(#4707)
```
Comment (StarpTech) [inline CHANGELOG.md:35]: The deprecation notes above aren't true anymore.

**Review by MrAlias — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (MrAlias) [reply CHANGELOG.md:36]: Done.

**Review by pellared — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (StarpTech) [reply CHANGELOG.md:36]: thank you.

**Pull request merged**